### PR TITLE
[XPU][TritonIntelGPU] Fix issue in convert DotOp A layout to LinearLayout of DPAS.

### PR DESCRIPTION
**Pull request by chengjunlu**
The code was using the old layout for DPAS operand A.

Update the code to the new layout which is supported by the OCL interface.

### CODE REVIEW

File: lib/Dialect/TritonGPU/IR/Dialect.cpp [hunk 1]
```diff
@@ -3217,6 +3213,10 @@ std::string getDistributedLayoutStr(RankedTensorType tensorType,
   int64_t tensorSize = product(tensorType.getShape());
   std::vector<std::string> elementMapping(tensorSize);
   std::vector<std::string> threadMapping;
+  unsigned threadsPerWarp = ll->getInDimSize(kLane);
```
Comment (chengjunlu) [inline lib/Dialect/TritonGPU/IR/Dialect.cpp:3216]: This changes is waiting to be merged to upstream Triton.
https://github.com/triton-lang/triton/pull/5196

File: third_party/intel/lib/Dialect/TritonIntelGPU/IR/LinearLayoutConversions.cpp [hunk 2]
```diff
@@ -341,38 +341,44 @@ LinearLayout combineCtaCgaWithShape(LinearLayout ctaLayout,
...
+// t16  t17  t18  t19  t20  t21  t22  t23  t24  t25  t26  t27  t28  t29  t30  t31   |
+// t0   t1   t2   t3   t4   t5   t6   t7   t8   t9   t10  t11  t12  t13  t14  t15   | M = 8 (repeat count)
+// t16  t17  t18  t19  t20  t21  t22  t23  t24  t25  t26  t27  t28  t29  t30  t31   |
+// t0   t1   t2   t3   t4   t5   t6   t7   t8   t9   t10  t11  t12  t13  t14  t15   |
+// t16  t17  t18  t19  t20  t21  t22  t23  t24  t25  t26  t27  t28  t29  t30  t31   v
 // In this case, the LinearLayout bases are:
-// Register:  {{0,1}, {4,0}}
-// Lane:      {{0,2}, {0,4}, {0,8}, {1,0}, {2,0}}
+// Register:  {{2,0}, {4,0}}
+// Lane:      {{0,1}, {0,2}, {0,4}, {0,8}, {2,0}}
```
Comment (jopperm) [inline third_party/intel/lib/Dialect/TritonIntelGPU/IR/LinearLayoutConversions.cpp:360]: I think it must be
```suggestion
// Lane:      {{0,1}, {0,2}, {0,4}, {0,8}, {1,0}}
```
to match the illustration above.

File: lib/Dialect/TritonGPU/IR/Dialect.cpp (hunk 1 above)
Comment (whitneywhtsang) [reply lib/Dialect/TritonGPU/IR/Dialect.cpp:3221]: Merging in https://github.com/intel/intel-xpu-backend-for-triton/pull/2783.

File: third_party/intel/lib/Dialect/TritonIntelGPU/IR/LinearLayoutConversions.cpp [hunk 3]
```diff
@@ -602,8 +614,7 @@ std::optional<LinearLayout>
 dotOperandDpasToLinearLayout(DotOperandEncodingAttr dotDpasLayout,
                              ArrayRef<int64_t> shape) {
   auto dpasLayout = cast<intel::DpasEncodingAttr>(dotDpasLayout.getParent());
-  if (dotDpasLayout.getOpIdx() == 0)
-    return std::nullopt;
+
```
Comment (chengjunlu) [inline third_party/intel/lib/Dialect/TritonIntelGPU/IR/LinearLayoutConversions.cpp:617]: Enable the DotOp A layout to LL layout conversion.

**Review by jopperm — APPROVED** [review]

**Pull request merged**